keps/sig-cli/kustomize-pod-spec-patch.md

```diff
@@ -0,0 +1,128 @@
+---
+title: Kustomize PodSpec Patch
+authors:
+ - "@pwittrock"
+owning-sig: sig-cli
+participating-sigs:
+ - sig-apps
+reviewers:
+ - "@Liujingfang1"
+ - "@janetkuo"
+approvers:
+ - "@monopole"
+editors:
+ - "@pwittrock"
+creation-date: 2019-03-01
+last-updated: 2019-03-01
+status: provisional
+see-also:
+replaces:
+superseded-by:
+ - n/a
+---
+
+# Kustomize PodSpec Patch
+
+## Table of Contents
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+  * [Goals](#goals)
+  * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+  * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+  * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Alternatives](#alternatives)
+
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+
+## Summary
+
+A common need for users is to inject sidecar containers across many different workloads.
+While overlays and patches allow injection on a per-Resource basis, they don't
+provide a mechanism for injecting the same patch across multiple workloads.
+
+## Motivation
+
+Cross-cutting PodSpec customizations are necessary for usecases such as:
```

Comment:
If current `patchesJson6902` or `patchesStrategicMerge` is extended to patch multiple targets (by regex in names or label selector), can we achieve the cross-cutting PodSpec customizations by them? If not, what is the missing piece?

Reply:
That SGTM.  I think that would be useful for stuff like ObjectMeta as well, which could be cross-cutting.  Would you be ok with me closing this KEP, can you could open up a new one proposing that solution?

Reply:
That sounds good to me. I can open a new one.